duckduckgo/iOS
# [DuckPlayer] - 3. URL management & FE comms

Task/Issue URL: https://app.asana.com/0/1204099484721401/1207658406526197/f

**Description**:
- Introduces a custom OmniBar icon to be used on specific cases (Youtube videos in this case)
- Updates the URL to properly display `duck://` addresses when DP is enabled
- Listen to AppSettings changes for DuckPlayer via `NotificationCenter` and pass them down to `DuckPlayer` and the UserScripts > FrontEnd via Combine
- Handle reloads when visiting a DuckPlayer page.
- Renames methods in DuckNavigationHandling for clarity
- @MainActorize all WebView navigation events
-  Updated tests



**Steps to test this PR**:
- [x] Disable Duck Player and confirm Youtube.com works as expected

1. Install custom privacy config from https://jsonblob.com/api/1253721903368888320
2. Go to Settings and switch DuckPlayer to 'Ask mode'
3. Go to a Youtube Video, and confirm the overlay shows up.
4. Tap Watch in Duck Player (Might need to tap twice quickly due to an FE bug)
- [x] Confirm video loads in DuckPlayer.
5. Go back
- [ ] Tap Watch here, and confirm video loads in DuckPlayer.

1. Go to DuckPlayer settings and enable 'Always' mode
2. Open a video or reload a video page if you have it open
- [x] Confirm video loads in Duck Player


1. While watching a video in Duck player, Hit 'reload' 
- [x]   Confirm that the video still shows in Duck Player
- [ ]   Confirm you see the DuckPlayer icon in the URL bar when a video is playing in DP
- [x]   Confirm the URL says "Duck Player"
- [x]   Click the URL bar to select the text
- [x]   Confirm the text turns to a valid duck:// URL

1. Enable Settings > Appearance > Show Full Site Address
- [x]  Go back to DuckPlayer and confirm the URL is now the full `duck://` address
